### PR TITLE
fix kubectl patch always show not patched

### PR DESCRIPTION
**Pull request by shiywang**

### CODE REVIEW

File: pkg/kubectl/cmd/patch.go [hunk 1]
```diff
@@ -200,14 +200,17 @@ func RunPatch(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 			}
 			count++
 
-			if err := info.Refresh(patchedObj, true); err != nil {
+			//info.Object is updated by the info.Refresh call.
+			//we should compute old data prior to refresh
+			oldData, err := json.Marshal(info.Object)
```
Comment (ymqytw) [inline pkg/kubectl/cmd/patch.go]: I have tried to move it here. But it doesn't work for all of the cases.
You should make sure all of these work as expected:
```
kubectl patch -f deployment.yaml -p '{}'
not patched
kubectl patch deployment nginx -p '{}'
not patched
kubectl patch -f deployment.yaml -p '{"spec":{"replicas":1}}'
patched
kubectl patch -f deployment.yaml -p '{"spec":{"replicas":2}}'
patched
```

Comment (shiywang) [inline pkg/kubectl/cmd/patch.go]: do you think code [here]( https://github.com/kubernetes/kubernetes/blob/90c12fff84f88cf560fd4e34add14bc79090b20c/pkg/kubectl/cmd/edit.go#L448 ) could works ?  if not, I think maybe we should also change here.

Comment (shiywang) [inline pkg/kubectl/cmd/patch.go]: @ymqytw ok, I'll try to test those scenarios you list today.

**Comment by shiywang** [comment]
@ymqytw close this since @liggitt  already fixed in https://github.com/kubernetes/kubernetes/pull/42391

**Pull request closed without merging**